### PR TITLE
fix spdlog cannot format an argument

### PR DESCRIPTION
**Pull request by Minghui-Gao**
added tvent type formattable to logging
fix spdlog cannot format an argument

### CODE REVIEW

**Comment by TejasJD** [comment]
Well, that solves the problem,
But I think we need to do this for all the classes we make then.

File: HorizonEngine/src/HorizonEngine/Logging/Logging.h [hunk 1]
```diff
@@ -24,6 +27,21 @@ namespace Hzn
...
 }
 
+template<>
+struct fmt::formatter<Hzn::Event> {
+	constexpr auto parse(format_parse_context& ctx) -> decltype(ctx.begin()) {
+		return ctx.end();
+	}
+	
+
+	template <typename FormatContext>
```
Comment (TejasJD) [inline HorizonEngine/src/HorizonEngine/Logging/Logging.h:37]: actually:
just writing
```
template<>
struct fmt::formatter<Hzn::Event> : fmt::ostream_formatter {};
```
this much does the job.

**Comment by Minghui-Gao** [comment]
-added event type formatter in Event.h
-fixed cannot log event

File: HorizonEngine/src/HorizonEngine/Logging/Logging.h (hunk 1 above)
Comment (Minghui-Gao) [reply HorizonEngine/src/HorizonEngine/Logging/Logging.h]: Thank you for your suggestion, this is a better solution.

**Pull request merged**